### PR TITLE
(DE-902) refactor(dbt): Migrate two tower queries to dbt

### PR DESCRIPTION
**Pull request by lmontier-pass**
# PR template

## Description

* Move sql queries for two tower DAG into dbt
* I Kept train/val/test queries in sql because 
  * More efficient to do that in Python rather than in SQL
  * Depends on parameters so not done in dbt

### Proof
image

## PR title format (except for MEP)

There is a linter on the PR title format. Please respect the following format:

<details>
<summary>(ticket) type(topic): comment</summary>

* ticket surrounded by parenthesis, with optionnaly a hyphen followed by one or more digits (e.g., -1234). The first part must be one of the following strings:
  * DA
  * DE
  * AE
  * DS
  * HF
  * BSR
  * PC

* type :
The second part to specify the type of change one of the following :
  * build
  * lint
  * ci
  * docs
  * feat
  * fix
  * perf
  * refactor
  * test
  * core
  * dbt

* topic within parenthesis: 1 word e.g., (dag)

* comment: tell us your life

examples:

* :white_check_mark: (DE-124) refactor(firebase): update source field
* :x: (DE-124) refactor (firebase): update source field **(space between type and topic)**
* :x: (DE-124) airflow(firebase): update source fiedd in DAG **(wrong type)**
* :x: (DE-124) (DE-124) refactor(firebase refacto): update source field **(topic in two words)**
* :white_check_mark: (BSR) docs(github): add PR title valid format in template

</details>

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/models/machine_learning/recommendation/ml_reco__training_data_enriched_click.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+select distinct
```
Comment (lrainteau-pass) [inline orchestration/dags/data_gcp_dbt/models/machine_learning/recommendation/ml_reco__training_data_enriched_click.sql:1]: Quelle est la différence entre ce modèle et le __ml_reco__training_data_click.sql__ ? C'est possible de les factoriser ? ou de spécifier la différence dans le nom du modèle : training_click_two_towers par exemple ?

File: orchestration/dags/jobs/ml/algo_training_two_towers.py [hunk 2]
```diff
@@ -180,77 +153,98 @@
...
                     "sourceTable": {
                         "projectId": GCP_PROJECT_ID,
-                        "datasetId": BIGQUERY_TMP_DATASET,
-                        "tableId": f"{DATE}_recommendation_{split}_data",
+                        "datasetId": BIGQUERY_ML_RECOMMENDATION_DATASET,
+                        "tableId": "training_data_booking",
                     },
                     "compression": None,
-                    "destinationUris": f"{dag_config['STORAGE_PATH']}/raw_recommendation_{split}_data/data-*.parquet",
+                    "destinationUris": f"{dag_config['STORAGE_PATH']}/bookings/data-*.parquet",
```
Comment (vbusson-pass) [inline orchestration/dags/jobs/ml/algo_training_two_towers.py:214]: relicat de test ?

File: orchestration/dags/jobs/ml/link_items.py [hunk 3]
```diff
@@ -68,7 +69,6 @@
 }
 
 SCHEDULE_DICT = {"prod": "0 4 * * 3", "stg": "0 6 * * 3", "dev": "0 6 * * 3"}
```
Comment (vbusson-pass) [inline orchestration/dags/jobs/ml/link_items.py:71]: schedule a migrer dans cron.py

**Review by dcuesta-pass — APPROVED** [review]

File: orchestration/dags/data_gcp_dbt/models/machine_learning/recommendation/ml_reco__training_data_enriched_click.sql (hunk 1 above)
Comment (lmontier-pass) [reply orchestration/dags/data_gcp_dbt/models/machine_learning/recommendation/ml_reco__training_data_enriched_click.sql]: regroupé en une seule query:)

File: orchestration/dags/jobs/ml/algo_training_two_towers.py (hunk 2 above)
Comment (lmontier-pass) [reply orchestration/dags/jobs/ml/algo_training_two_towers.py:214]: non c'est l'affichage de git le problème

File: orchestration/dags/jobs/ml/link_items.py (hunk 3 above)
Comment (lmontier-pass) [reply orchestration/dags/jobs/ml/link_items.py]: done

**Review by lrainteau-pass — APPROVED** [review]

**Pull request merged**